src/main/java/com/snowflake/kafka/connector/internal/streaming/SnowflakeSinkServiceV2.java

```diff
@@ -0,0 +1,426 @@
+  private boolean enableCustomJMXMonitoring = SnowflakeSinkConnectorConfig.JMX_OPT_DEFAULT;
+
+  // default is at_least_once semantic (To begin with)
+  // TODO: SNOW-526435
+  private SnowflakeSinkConnectorConfig.IngestionDeliveryGuarantee ingestionDeliveryGuarantee =
+      SnowflakeSinkConnectorConfig.IngestionDeliveryGuarantee.AT_LEAST_ONCE;
+
+  // ------ Streaming Ingest ------ //
+  // needs url, username. p8 key, role name
+  private final SnowflakeStreamingIngestClient streamingIngestClient;
+
+  // Config set in JSON
+  private final Map<String, String> connectorConfig;
+
+  private final String taskId;
+
+  private final String streamingIngestClientName;
+
+  /**
+   * Key is formulated in {@link #partitionChannelKey(String, int)} }
+   *
+   * <p>value is the Streaming Ingest Channel implementation (Wrapped around TopicPartitionChannel)
+   */
+  private final Map<String, TopicPartitionChannel> partitionsToChannel;
+
+  public SnowflakeSinkServiceV2(
+      SnowflakeConnectionService conn, Map<String, String> connectorConfig) {
+    if (conn == null || conn.isClosed()) {
+      throw SnowflakeErrors.ERROR_5010.getException();
+    }
+
+    this.fileSize = SnowflakeSinkConnectorConfig.BUFFER_SIZE_BYTES_DEFAULT;
+    this.recordNum = SnowflakeSinkConnectorConfig.BUFFER_COUNT_RECORDS_DEFAULT;
+    this.flushTime = SnowflakeSinkConnectorConfig.BUFFER_FLUSH_TIME_SEC_DEFAULT;
+    this.conn = conn;
+    this.recordService = new RecordService();
+    this.telemetryService = conn.getTelemetryClient();
+    this.topic2TableMap = new HashMap<>();
+
+    // Setting the default value in constructor
+    // meaning it will not ignore the null values (Tombstone records wont be ignored/filtered)
+    this.behaviorOnNullValues = SnowflakeSinkConnectorConfig.BehaviorOnNullValues.DEFAULT;
+
+    this.connectorConfig = connectorConfig;
+    // Streaming ingest requires a properties.
+    // TODO request a builder pattern for SI Client.
+    Properties streamingClientProps = new Properties();
+    streamingClientProps.putAll(connectorConfig);
+    this.taskId = connectorConfig.getOrDefault(Utils.TASK_ID, "-1");
+    this.streamingIngestClientName =
+        STREAMING_CLIENT_PREFIX_NAME + conn.getConnectorName() + "_" + taskId;
+    this.streamingIngestClient =
+        SnowflakeStreamingIngestClientFactory.builder(this.streamingIngestClientName)
+            .setProperties(streamingClientProps)
+            .build();
+    this.partitionsToChannel = new HashMap<>();
+  }
+
+  @Override
+  public void startTask(String tableName, String topic, int partition) {
+    String partitionChannelKey = partitionChannelKey(topic, partition);
+    LOGGER.info(
+        "Opening a channel with name:{} for table name:{}, topic:{}, partition:{}",
+        partitionChannelKey,
+        tableName,
+        topic,
+        partition);
+    // the table should be present before opening a channel so lets do a table existence check here
+    createTableIfNotExists(tableName);
```

Comment:
This is interesting, how would you create the table since you don't have the schema, do you?

Reply:
For now we decided we will go with just the two column schema right? Same as what we have right now.. 
(metadata and content both being variant columns)

Reply:
Oh, i see, I think we just want to error out if the table doesn't exist, otherwise it feels like a regression from the customer's point after we have schematization

Reply:
I think if table doesnt exist, it's a cleaner approach. Sorry, I dont understand how it will be a regression.

Reply:
Created a JIRA to discuss this further. SNOW-536661